src/main/java/core/basesyntax/bookstore/repository/user/UserRepository.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.bookstore.repository.user;
+
+import core.basesyntax.bookstore.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("SELECT u FROM User u LEFT JOIN FETCH u.roles WHERE u.email = :email")
+    Optional<User> findByEmail(String email);

```

Comment:
I would explicitly specify that this method fetches roles, it would be easier to maintain later.
```suggestion
    Optional<User> findByEmailFetchRoles(String email);

```